spec/controllers/subjects_controller_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+include Warden::Test::Helpers
+Warden.test_mode!
+
+RSpec.describe SubjectsController, type: :controller do
+  render_views
+
+  describe 'GET #new' do
+    before do
+      allow(controller).to receive(:user_signed_in?).and_return(false)
+      allow(controller).to receive(:current_user).and_return(false)
+    end
+
+    context 'when user is unauthenticated' do
+      it 'redirects to login' do
+        get :new
+        expect(response).to redirect_to unauthenticated_root_url
+      end
+    end
+
+    context 'when user is authenticated' do
+      it 'assigns @subject to new' do
+        allow(controller).to receive(:user_signed_in?).and_return(true)
+        get :new
+        expect(assigns(:subject)).to be_a_new(Subject)
+      end
+
+      it "renders the subject template" do
+        allow(controller).to receive(:user_signed_in?).and_return(true)
+        get :new
+        expect(response).to render_template('subjects/new')
+      end
+    end
+
+    let(:user) { FactoryBot.create(:user) }
+    before { login_as(user, scope: :user) }
+
+    before do
+      visit 'subjects/new'
+      fill_in 'subject[title]', with: 'Subject Title'
+      fill_in 'subject[body]', with: 'In subject body. This must be longer then 20 symbols minimum'
+    end
+
+
+    context 'when user saves subject' do
+      it "creates the subject and redirect to subject view" do
+        find('[name=commit]').click
+        current_path.should eq '/subjects/1'
+      end
+
+      it "adds subject in subjects table" do
+        expect{ find('[name=commit]').click }.to change(Subject, :count).by(1)
+      end
+    end
+
+    context 'when user fails to save subject' do
+      it 'title is too short' do
```

Comment:
tests is your documentation so it would be nice if you could read them like a normal english sentences. For example this particular test will be converted to sentence:
```
SubjectsController GET #new
  when user fails to save subject 
    title is too short
``` 
which is not very human readable. On the other hand you have quite fluent tests in 47 or 52 lines: 
```
SubjectsController GET #new 
  when user saves subject 
    creates the subject and redirect to subject view
    adds subject in subjects table`
```
try running your specs with `documentation` formatter:
```
rspec -f documentation
```

and you will see more examples

Reply:
Oh, okay, I will keep that in mind next time, thanks for good advice!

Reply:
fixed